<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MaterialEditorWindow</class>
 <widget class="QMainWindow" name="MaterialEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>781</width>
    <height>600</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Segoe UI</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>Material editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="0">
     <widget class="QPushButton" name="addVariable">
      <property name="toolTip">
       <string>Add variable to be exposed in content browser.</string>
      </property>
      <property name="text">
       <string>Add variable...</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="addPass">
      <property name="toolTip">
       <string>Add new pass to this material. Passes dictate when to render objects using this material.</string>
      </property>
      <property name="text">
       <string>Add pass...</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="removePass">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>Remove passes selected in list.</string>
      </property>
      <property name="text">
       <string>Remove selected</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QPushButton" name="removeVariable">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>Remove variables so that they are not exposed in the content browser.</string>
      </property>
      <property name="text">
       <string>Remove selected</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="Line" name="line">
      <property name="styleSheet">
       <string notr="true">QFrame{ color:rgb(95, 95, 95); }</string>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="2">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="4" column="0" colspan="3">
       <widget class="Line" name="line_3">
        <property name="styleSheet">
         <string notr="true">QFrame{ color:rgb(95, 95, 95); }</string>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0" rowspan="3" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Material</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="materialBox"/>
        </item>
        <item>
         <widget class="QPushButton" name="newMaterial">
          <property name="text">
           <string>Create new</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteMaterial">
          <property name="text">
           <string>Delete selected</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="2" rowspan="4">
       <widget class="QLabel" name="materialThumbnail">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>67</width>
          <height>67</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">border: 2px solid gray;
border-radius: 5px;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLineEdit" name="materialName"/>
        </item>
        <item>
         <widget class="QCheckBox" name="isVirtual">
          <property name="text">
           <string>Virtual</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="descEditor">
          <property name="text">
           <string>Edit description...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Inherit:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="inherit">
          <property name="toolTip">
           <string>Materials to inherit, separated by | characters. Can only inherit materials from the toolkit base.xml file.</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="QTableWidget" name="variableWidget">
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="cornerButtonEnabled">
       <bool>false</bool>
      </property>
      <property name="columnCount">
       <number>7</number>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Type</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Default</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Min</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Max</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Description</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string/>
       </property>
      </column>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QTableWidget" name="passWidget">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="cornerButtonEnabled">
       <bool>false</bool>
      </property>
      <property name="columnCount">
       <number>4</number>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Batch</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Shader</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Variation</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string/>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>781</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_list"/>
    <addaction name="actionOpen_list"/>
    <addaction name="separator"/>
    <addaction name="actionSave_list"/>
    <addaction name="actionSave_list_as"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen_list">
   <property name="text">
    <string>Open list...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionNew_list">
   <property name="text">
    <string>New list...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionSave_list">
   <property name="text">
    <string>Save list</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_list_as">
   <property name="text">
    <string>Save list as...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
